internal/beater/beater_test.go

```diff
@@ -76,6 +76,9 @@ func TestStoreUsesRUMElasticsearchConfig(t *testing.T) {
 	)
 	require.NoError(t, err)
 	defer cancel()
+
+	fetcher.Wait()
```

Comment:
As mentioned elsewhere, we should not be adding methods to the interface just for tests. This complicates all of the implementations.

What I suggest we do instead is just create the fetcher, but don't invoke it. It should poll the test server in the background. Then in the test server, send to a channel rather than setting `called = true`, and wait for that channel to be signalled here.

I don't think we need to invoke the fetcher here. The basic functionality is tested in internal/sourcemap.

Reply:
Updated :+1: